risc0/r0vm/src/actors/mod.rs

```diff
@@ -67,69 +68,18 @@ use self::{
     worker::Worker,
 };
 
-#[derive(Debug, Serialize, Deserialize)]
-pub(crate) struct WorkerConfig {
-    pub pools: Vec<PoolConfig>,
-}
-
-impl From<Vec<TaskKind>> for WorkerConfig {
-    fn from(task_kinds: Vec<TaskKind>) -> Self {
-        WorkerConfig {
-            pools: vec![PoolConfig {
-                count: 1,
-                profile: None,
-                task_kinds,
-            }],
-        }
-    }
-}
-
-#[derive(Clone, Debug, Serialize, Deserialize)]
-pub(crate) struct PoolConfig {
-    pub count: usize,
-    pub profile: Option<DevModeDelay>,
-    pub task_kinds: Vec<TaskKind>,
-}
-
 #[tokio::main]
-pub(crate) async fn async_main(args: &Cli) -> Result<(), Box<dyn StdError>> {
-    let is_manager = args.mode.manager;
-
-    let config = if let Some(ref path) = args.mode.config {
+pub(crate) async fn async_main(config_path: Option<PathBuf>) -> Result<(), Box<dyn StdError>> {
+    let config: AppConfig = if let Some(ref path) = config_path {
         let str = tokio::fs::read_to_string(path).await?;
-        Some(toml::from_str(&str)?)
+        toml::from_str(&str)?
```

Comment:
something like this maybe? I didn't test it.
```suggestion
        let table: toml::Table = toml::from_str(&str)?;
        let version_enrty = table.get("version").ok_or_else(|| anyhow!("missing version"))?;
        let version = version_entry.as_integer().ok_or_else(|| anyhow!("version not integer"))?;
        ensure!(version == 1, "version {version} not supported);
        toml::from_str(&str)?
```